### PR TITLE
TINY-8763: The value of togglable elements was not reset if not defined in the definition

### PR DESCRIPTION
**Pull request by tiny-ben-tran**
Related Ticket: TINY-8763

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/alloy/src/test/ts/atomic/dom/ReconcileTest.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    const definition = {
+      ...elementDefinition,
+      value: Optional.from('abc')
+    };
+    const newDom = reconcileToDom(definition, component);
+    assert.equal((newDom.dom as HTMLInputElement).value, 'abc');
+  });
+
+  it('TINY-8736: Default value of element to empty string if value is not specified in the definition', () => {
+    const component = SugarElement.fromHtml<HTMLInputElement>('<textarea>hi</textarea/>');
```
Comment (spocke) [inline modules/alloy/src/test/ts/atomic/dom/ReconcileTest.ts:32]:
```suggestion
    const component = SugarElement.fromHtml<HTMLTextAreaElement>('<textarea>hi</textarea>');
```

File: modules/alloy/src/test/ts/atomic/dom/ReconcileTest.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    const newDom = reconcileToDom(elementDefinition, component);
+    assert.equal((newDom.dom as HTMLInputElement).value, '');
+  });
+
+  it('TINY-8736: Default the modified value of element to empty string if value is not specified in the definition', () => {
+    const component = SugarElement.fromHtml<HTMLTextAreaElement>('<textarea>How</textarea>');
+    component.dom.value = 'hello';
```
Comment (spocke) [inline modules/alloy/src/test/ts/atomic/dom/ReconcileTest.ts:39]: Not sure why this is in there the value is already set to How.

File: modules/alloy/src/test/ts/atomic/dom/ReconcileTest.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      value: Optional.from('abc')
+    };
+    const newDom = reconcileToDom(definition, component);
+    assert.equal((newDom.dom as HTMLInputElement).value, 'abc');
+  });
+
+  it('TINY-8736: Default value of element to empty string if value is not specified in the definition', () => {
+    const component = SugarElement.fromHtml<HTMLInputElement>('<textarea>hi</textarea/>');
+    const newDom = reconcileToDom(elementDefinition, component);
+    assert.equal((newDom.dom as HTMLInputElement).value, '');
```
Comment (spocke) [inline modules/alloy/src/test/ts/atomic/dom/ReconcileTest.ts:34]:
```suggestion
    assert.equal((newDom.dom as HTMLTextAreaElement).value, '');
```

**Review by spocke — APPROVED** [review]

File: modules/alloy/src/test/ts/atomic/dom/ReconcileTest.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  it('TINY-8736: Default the value of element to empty string if not defined in the definition', () => {
+    const component = SugarElement.fromHtml<HTMLTextAreaElement>('<textarea>hi</textarea/>');
+    const newDom = reconcileToDom(elementDefinition, component);
+    assert.equal((newDom.dom as HTMLTextAreaElement).value, '');
+  });
+
+  it('TINY-8736: Should not update value of a non togglable element', () => {
+    const component = SugarElement.fromHtml<HTMLDivElement>('<div>wow</div>');
+    assert.equal((component.dom as any).value, undefined);
```
Comment (lnewson) [inline modules/alloy/src/test/ts/atomic/dom/ReconcileTest.ts:39]:
```suggestion
    assert.isUndefined((component.dom as any).value);
```

File: modules/alloy/src/test/ts/atomic/dom/ReconcileTest.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    const newDom = reconcileToDom(elementDefinition, component);
+    assert.equal((newDom.dom as any).value, undefined);
```
Comment (lnewson) [inline modules/alloy/src/test/ts/atomic/dom/ReconcileTest.ts:41]:
```suggestion
    assert.isUndefined((newDom.dom as any).value);
```

**Review by lnewson — APPROVED** [review]

File: modules/alloy/src/test/ts/atomic/dom/ReconcileTest.ts (hunk 2 above)
Comment (tiny-ben-tran) [reply modules/alloy/src/test/ts/atomic/dom/ReconcileTest.ts]: It is a leftover from an experimentation

**Pull request merged**